mobilecoind-dev-faucet/src/bin/main.rs

```diff
@@ -8,18 +8,39 @@
 use clap::Parser;
 use mc_common::logger::{create_app_logger, log, o};
 use mc_mobilecoind_dev_faucet::{data_types::*, Config, State};
-use rocket::{get, post, routes, serde::json::Json};
+use rocket::{get, post, routes, serde::json::Json, Shutdown};
 
 /// Request payment from the faucet, and map the rust result onto json for
 /// rocket appropriately
-#[post("/", format = "json", data = "<req>")]
+#[post("/", data = "<req>")]
 async fn post(
     state: &rocket::State<State>,
     req: Json<JsonFaucetRequest>,
 ) -> Json<JsonSubmitTxResponse> {
     Json(state.handle_post(&req).await.into())
 }
 
+/// Request to initiate a slam, and map the rust result onto json for
+/// rocket appropriately
+#[post("/slam", data = "<req>")]
+async fn post_slam(
+    state: &rocket::State<State>,
+    req: Option<Json<JsonSlamRequest>>,
+    shutdown: Shutdown,
+) -> Json<JsonSlamResponse> {
+    let req: JsonSlamRequest = if let Some(val) = req {
```

Comment:
Might be able to do something like `req.map(|r| *r).unwrap_or_default()`

Reply:
thanks will try